website/static/js/registrationUtils.js

```diff
@@ -0,0 +1,914 @@
+require('css/registrations.css');
+
+var $ = require('jquery');
+var ko = require('knockout');
+var bootbox = require('bootbox');
+var moment = require('moment');
+var URI = require('URIjs');
+
+require('js/koHelpers');
+
+var $osf = require('js/osfHelpers');
+var language = require('js/osfLanguage').registrations;
+
+var editorExtensions = require('js/registrationEditorExtensions');
+
+var currentUser = window.contextVars.currentUser || {
+    id: null,
+    name: 'Anonymous'
+};
+
+/**
+ * @class Comment
+ * Model for storing/editing/deleting comments on form fields
+ *
+ * @param {Object} data: optional data to instatiate model with
+ * @param {User} data.user
+ * @param {Date} data.lastModified
+ * @param {String} data.value
+ *
+ * @type User
+ * @property {String} id
+ * @property {String} name
+ **/
+function Comment(data) {
```

Comment:
I'm going to pass on reviewing a substantial portion of the commenting and file-picking code, insofar as @HarryRybacki indicates that it's not active/ready to review yet.